From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Mon, 28 Oct 2019 09:44:36 -0700
Subject: ANDROID: fix VIDEOBUF2_CORE dependency in 'allmodconfig' builds

Fix missing dependency for CONFIG_VIDEOBUF2_CORE for
'allmodconfig' builds. Since CONFIG_MEDIA_SUPPORT is
always selected, drop it from gki_defconfig

Fixes: 649238947d688 ("ANDROID: init: GKI: enable hidden configs
for media")

Test: successful 'allmodconfig' build
Change-Id: I1dd5ff154c1aeb90457c23dc233cb32595bd9bed
Signed-off-by: Todd Kjos <tkjos@google.com>
---
 arch/arm64/configs/gki_defconfig | 1 -
 arch/x86/configs/gki_defconfig   | 1 -
 init/Kconfig.gki                 | 1 +
 3 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 74858518d011..0933c134a2d0 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -298,7 +298,6 @@ CONFIG_WATCHDOG=y
 CONFIG_MFD_ACT8945A=y
 CONFIG_MFD_SYSCON=y
 CONFIG_REGULATOR=y
-CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_MEDIA_CONTROLLER=y
 # CONFIG_VGA_ARB is not set
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index ba629336a4f2..277b0069671e 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -257,7 +257,6 @@ CONFIG_GPIOLIB=y
 CONFIG_DEVFREQ_THERMAL=y
 # CONFIG_X86_PKG_TEMP_THERMAL is not set
 CONFIG_REGULATOR=y
-CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_DRM=y
 # CONFIG_DRM_FBDEV_EMULATION is not set
diff --git a/init/Kconfig.gki b/init/Kconfig.gki
index c9505a791280..c5a080e7784b 100644
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -56,6 +56,7 @@ config GKI_HIDDEN_QCOM_CONFIGS
 config GKI_HIDDEN_MEDIA_CONFIGS
 	bool "Hidden Media configs needed for GKI"
 	select VIDEOBUF2_CORE
+	select MEDIA_SUPPORT
 	help
 	  Dummy config option used to enable hidden media configs.
 	  These are normally selected implicitely when a module
